nabs/preprocessors.py

```diff
@@ -301,4 +302,64 @@ def daq_during_plan():
     return (yield from daq_during_plan())
 
 
+def step_size_decorator(plan):
+    """
+    Grab the last argument (number of steps), and intepret as
+    - step size if float
+    - number of steps if integer
+
+    Only works on step scans in one dimension.
+
+    Parameters
+    ----------
+    plan : plan
+        A bluesky plan that yields bluesky Msg objects.  Must be
+        a scan in one dimension, with the last argument being
+        the number of scan points / step size.
+
+    Returns
+    -------
+    step_size_plan : plan
+        The same plan as before, but modified appropriately to
+        differentiate between step size and number of steps.
+        This will be a callable generator function.
+    """
+
+    @wraps(plan)
+    def inner(*args, **kwargs):
+        if 'num' in kwargs:
+            # Currently unneeded, since daq_ascan and daq_dscan
+            # do not support num kwarg
+            n = kwargs.pop('num')
+        else:
+            # assumes (det_list, motor, mmin, mmax, num)
+            n = args[-1]
+
+        if type(n) not in [int, float]:
+            raise TypeError("Step size / number of steps is "
+                            "neither float nor integer")
+
+        if type(n) is int:
+            # interpret as number of steps (default)
+            yield from plan(*args, **kwargs)
+        elif type(n) is float:
+            # interpret as step size
+            mmin, mmax = args[2], args[3]
+            if n > (mmax-mmin):
+                raise ValueError(f"Step size provided {n} greater "
+                                 "than the range provided "
+                                 f"{mmax-mmin}")
+            step_list = list(np.arange(mmin, mmax, n))
+            # new endpoint needed, for cases where
+            # (range % step_size) != 0 or to include endpoint
+            if np.isclose(step_list[-1] + n, mmax):
+                step_list.append(step_list[-1] + n)
+            n_steps = len(step_list)
```

Comment:
Might be a good idea to raise an exception on `n_steps == 0` to catch any other weird shaping here